server/service/src/sync/translations/master_list.rs

```diff
@@ -47,6 +49,47 @@ impl SyncTranslation for MasterListTranslation {
         };
         Ok(PullTranslateResult::upsert(result))
     }
+
+    // Soft deletes were implemented in OG months after program requisitions was
+    // rolled out, so previously hard deleted records may be gone even if they
+    // are linked to program. Recreate these records to keep the FK link.
+    fn try_translate_from_delete_sync_record(
+        &self,
+        connection: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<PullTranslateResult, anyhow::Error> {
+        let program =
+            ProgramRowRepository::new(connection).find_one_by_id(&sync_record.record_id)?;
+        let master_list =
+            MasterListRowRepository::new(connection).find_one_by_id(&sync_record.record_id)?;
+
+        if let Some(program) = program {
+            let result = MasterListRow {
+                id: program.id.clone(),
+                name: "missing_program".to_string(),
+                code: "".to_string(),
+                // program name is the master list description
+                description: program.name.clone(),
+                is_active: false,
+            };
+            return Ok(PullTranslateResult::upsert(result));
+        }
+
+        if let Some(master_list) = master_list {
```

Comment:
Put this in for master lists we already have that were deleted before the soft delete was implemented in OG, but maybe can just hard delete them since they aren't linked to any requisition programs?